common/desktop-exports

```diff
@@ -125,15 +125,19 @@ prepend_dir __EGL_VENDOR_LIBRARY_DIRS "/var/lib/snapd/lib/glvnd/egl_vendor.d"
 append_dir __EGL_VENDOR_LIBRARY_DIRS "$SNAP_DESKTOP_RUNTIME/usr/share/glvnd/egl_vendor.d"
 
 # Tell Pipewire where to find its configuration and plugins
-export PIPEWIRE_CONFIG_DIR="$SNAP_DESKTOP_RUNTIME/usr/share/pipewire"
-export PIPEWIRE_MODULE_DIR="$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/pipewire-0.3"
-export SPA_PLUGIN_DIR="$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/spa-0.2"
+if [[ -z ${PIPEWIRE_CONFIG_DIR} || -z ${PIPEWIRE_MODULE_DIR} || -z ${SPA_PLUGIN_DIR} ]]; then
```

Comment:
Each path variable has to be checked individually and not collectively as here. I.e., we don't want PIPEWIRE_MODULE_DIR overwritten just because PIPEWIRE_CONFIG_DIR is missing.

Reply:
Yes, we do. Specifically for Pipewire, the config file, or the module dir, or the spa plugin dir, all need to be set. If not there might be api mismatch. This is the best way to keep both the installation of gstreamer and pipewire separated. I have been using these with `kooha` and `clapper` and they both are working perfectly.

Reply:
Indeed, might is the best word: Chromium, for one, set PIPEWIRE_MODULE_DIR and SPA_PLUGIN_DIR without setting PIPEWIRE_CONFIG_DIR and did fine that way.

We should not punish maintainers that deliberately set only a subset of those variables by _silently overruling their explicit intent_. This whole thing is not trivial to track down already (well, I can talk for myself) with all the various scattered components involved. I'd much rather respect their decision and let them cope with the consequences of it.

Reply:
It had PIPEWIRE_CONFIG_NAME, not _DIR, which is being checked here.

I think exiting the execution is too drastic. Couldn't we just allow the snap to run with each variable it defined and otherwise supply the sane defaults here? As in
```
export PIPEWIRE_CONFIG_DIR=${PIPEWIRE_CONFIG_DIR:-"$SNAP_DESKTOP_RUNTIME/usr/share/pipewire"}
export PIPEWIRE_MODULE_DIR=${PIPEWIRE_MODULE_DIR:-"$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/pipewire-0.3"}
export SPA_PLUGIN_DIR=${SPA_PLUGIN_DIR:-"$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/spa-0.2"}
```

Reply:
Yup, I think this is the perfect solution.

Reply:
Done!